filters/tracing/statebagtotag.go

```diff
@@ -53,16 +53,22 @@ func NewStateBagToTag() filters.Spec {
 	return stateBagToTagSpec{}
 }
 
-func (f stateBagToTagFilter) Request(ctx filters.FilterContext) {
+func (f *stateBagToTagFilter) Request(ctx filters.FilterContext) {
+	value, ok := ctx.StateBag()[f.stateBagItemName]
+	if !ok {
+		return
+	}
+
 	span := opentracing.SpanFromContext(ctx.Request().Context())
 	if span == nil {
 		return
 	}
-	value, ok := ctx.StateBag()[f.stateBagItemName]
-	if !ok {
-		return
+
+	if _, ok := value.(string); ok {
+		span.SetTag(f.tagName, value)
```

Comment:
See https://github.com/golang/go/issues/64541 why not
```go
if s, ok := value.(string); ok {
    span.SetTag(f.tagName, s)
}
```

Reply:
I guess we have similar cases all over the place.

Reply:
I checked all span.SetTag callsites and its fine. There may be other places that receive `interface{}` like SetTag of course but I think this is a very specific pattern here - optimizing for string case.

We could simply use `span.SetTag(f.tagName, value)` without type assertion but that is not guaranteed to work for types other than string, bool and ints:
```go
// Adds a tag to the span.
	//
	// If there is a pre-existing tag set for `key`, it is overwritten.
	//
	// Tag values can be numeric types, strings, or bools. The behavior of
	// other tag value types is undefined at the OpenTracing level. If a
	// tracing system does not know how to handle a particular value type, it
	// may ignore the tag, but shall not panic.
	//
	// Returns a reference to this Span for chaining.
	SetTag(key string, value interface{}) Span
```
so I decided to keep fmt.Sprint()